processor/lookupprocessor/README.md

```diff
@@ -0,0 +1,47 @@
+# Lookup Processor
+This processor is used to lookup values in a csv file.
+
+## Supported pipelines
+- Logs
+- Metrics
+- Traces
+
+## How It Works
+1. This processor will load and periodically refresh a CSV file in memory.
+2. When telemetry is received, the processor checks if the configured `field` exists in the configured `context`.
+3. If the field exists and the CSV contains a matching value, all other values in that record are added to the `context` of the telemetry. The header of each value is used when adding these fields.
+
+## Configuration
+| Field        | Type     | Default | Description |
+| ---          | ---      | ---     | ---         |
+| csv          | string   | ` `     | The location of the CSV file used for lookups. The processor will periodically reload this in memory every minute. |
+| context      | string   | ` `     | The context of the telemetry to check and use when performing lookups. Supported values are `attributes`, `body`, `resource`. |
+| field        | string   | ` `     | The field to match when performing a lookup. For a lookup to succeed, the field name and value must be the same in the CSV file and telemetry. |
```

Comment:
Is there any way we can use OTTL here for context instead? e.g. context is a OTTL path to a pdata.Map instead, so it can support nested attributes and the like.

Reply:
Asking because we've been wanting to use OTTL as much as possible instead of defining our own systems for specifying paths in telemetery, for consistency.

Reply:
I considered OTTL, but wanted to avoid making this processor too complicated until we've proved out the customer use case first. I tried to architect the config in such a manner that we could adopt OTTL later if it's necessary. At the moment, the simplicity of this design was intentional to avoid getting bogged down in edge cases and get this in a customer's hands first.

Reply:
However, if you think it's extremely easy to add, I would not turn down a code snippet/commit showing how to do it. I just wanted to avoid getting too wrapped up in less likely use cases, until we've gotten feedback.

Reply:
Eh, I think it would take a bit of refactor, might end up simplifying the code a little but maybe it's not worth the effort right now.

Suggestion instead then; Can we change `resource` to `resource.attributes` so that if we DO want to make the context field OTTL in the future, it's not a breaking change?

Reply:
Yeah, I think that's a reasonable change to make. Since we'll be wrapping this with a BindPlane processor, we can just show it as `resource` there.